### DISCUSSION TITLE
Plugin Store

### DISCUSSION THREAD

**Discussion by onesounds** [Flow-Launcher/Flow.Launcher#732] [Ideas]
![image](https://user-images.githubusercontent.com/6903107/136307025-9c7d05aa-9ac3-4d20-9eb6-f3a97253c33a.png)

~~Load https://github.com/Flow-Launcher/Flow.Launcher.PluginsManifest when open setting,~~
- Load https://github.com/Flow-Launcher/Flow.Launcher.PluginsManifest when choose plugin store tab.
- Managing in setting window.
- user can found easily new plugin and updates.
- The command line installation method (pm install) remains the same.

**Why**
- Installable, all plug-in lists should be visible at a glance.

**Additional considerations**
- This may require a refresh button. (list update check)
- What if there are more than 30 plug-ins? Wouldn't there be a heavy load?
- It would be nice if you could receive icons when you receive the list.

**ETC**
- I considered the idea of integrating with the plug-in tab, but we'd better distinguish between setting management/installation.

Done. https://github.com/Flow-Launcher/Flow.Launcher/pull/740

**Comment by Zeroto521** [comment]
That's a good idea for users.
Easily know and get plugins, no more type command just browse and click.

I just have a problem with the store.
For now, we only have 20+ plugins. Such a number of plugins could be list here. But when the number grows up, how to find a plugin that you want would be a problem.

**Reply by onesounds** [reply]
Good Point, There should be page control.
Or I think it would be okay to make it first, and make it again when it exceeds 50. 😂

**Reply by deefrawley** [reply]
What about a search function here?

**Reply by onesounds** [reply]
> What about a search function here?

I don't think it's essential, but I think it'd be nice to have it.

**Reply by red9350** [reply]
The plugin store could have 2 tabs: 
"Installed" only shows the currently installed plugins, and has a "Check for updates" button;
"Available" shows all the plugins present in the repo, minus the ones already installed;

Both tabs should have a search bar

**Comment by taooceros** [comment]
@jjw24 should we migrate some functionality in pluginsmanager plugin to built-in project? like the pluginsmanifest, and let it be accessed via ipublic api or something else.

**Reply by jjw24** [reply]
I originally wrote the pluginsmanager with the intent to build the store within the plugin itself but this is fine as well since this will be a dedicated section. I think in this case we might move PluginsManager into flow as a class rather than a plugin because I dont think it's a good idea to expose the plugin manifest and allow the usage of functionalities such as installation other than by Flow itself.

**Reply by jjw24** [reply]
lets just convert PluginsManager into the Plugin Store's backend code, this is probably the easiest

**Reply by jjw24** [reply]
hmm do we then lose the ability to use it via the query?

**Reply by taooceros** [reply]
> hmm do we then lose the ability to use it via the query?

Not really. We can still have a plugin project link to the manifest project I guess.

**Reply by jjw24** [reply]
Sorry what do you mean by link?

**Reply by taooceros** [reply]
Oh I mean we can let the plugin one reference to the built-in project.

**Comment by onesounds** [comment]
https://github.com/Flow-Launcher/Flow.Launcher/pull/740 done
